### PR TITLE
Doc: How to search for users in multiple LDAP groups

### PR DESCRIPTION
**Pull request by jifox**
Extend <https://nautobot.readthedocs.io/en/latest/configuration/authentication/ldap/#user-authentication> documentation to show how to search multiple LDAP Groups for users.

## Added text:

The following snippet shows how to search for users in multiple LDAP groups:

- Define the user-groups in *.env file (delimiter `';'`)

    ```bash
    # Groups to search for user objects. "(sAMAccountName=%(user)s),..."
    NAUTOBOT_AUTH_LDAP_USER_SEARCH_DN=OU=IT-Admins,OU=special-users,OU=Acme-User,DC=Acme,DC=local;OU=Infrastruktur,OU=IT,OU=my-location,OU=User,OU=Acme-User,DC=Acme,DC=local
    ```

- Import LDAPSearchUnion in `nautobot_config.py`

    ```python
    from django_auth_ldap.config import ..., LDAPSearchUnion
    ```

- In `nautobot_config.py` replace the AUTH_LDAP_USER_SEARCH command from above with:

    ```bash
    user_search_dn_list = str(AUTH_LDAP_USER_SEARCH_DN).split(";")
    ldapsearch_objects = []
    for sdn in user_search_dn_list:
        ldapsearch_objects.append(LDAPSearch(sdn.strip(), ldap.SCOPE_SUBTREE, "(sAMAccountName=%(user)s)"))
    AUTH_LDAP_USER_SEARCH = LDAPSearchUnion(*ldapsearch_objects)
    ```

### CODE REVIEW

File: nautobot/docs/configuration/authentication/ldap.md [hunk 1]
```diff
@@ -126,6 +126,31 @@ AUTH_LDAP_USER_ATTR_MAP = {
 }
 ```
 
+The following snippet shows how to search for users in multiple LDAP groups:
```
Comment (glennmatthews) [inline nautobot/docs/configuration/authentication/ldap.md:129]:
```suggestion
#### Searching in Multiple LDAP Groups
```

File: nautobot/docs/configuration/authentication/ldap.md [hunk 2]
```diff
@@ -126,6 +126,31 @@ AUTH_LDAP_USER_ATTR_MAP = {
...
+
+- In `nautobot_config.py` replace the AUTH_LDAP_USER_SEARCH command from above with:
+
+    ```bash
+    user_search_dn_list = str(AUTH_LDAP_USER_SEARCH_DN).split(";")
+    ldapsearch_objects = []
+    for sdn in user_search_dn_list:
+        ldapsearch_objects.append(LDAPSearch(sdn.strip(), ldap.SCOPE_SUBTREE, "(sAMAccountName=%(user)s)"))
+    AUTH_LDAP_USER_SEARCH = LDAPSearchUnion(*ldapsearch_objects)
+    ```
```
Comment (glennmatthews) [inline nautobot/docs/configuration/authentication/ldap.md:152]: This doesn't render properly [on ReadTheDocs](https://nautobot--1080.org.readthedocs.build/en/1080/configuration/authentication/ldap/#user-authentication) (you can check the rendering locally with `mkdocs serve` then pointing your browser to `http://localhost:8001`)

Possibly the following might render correctly, but please try it out:

````suggestion
Define the user-groups in your environment, such as a *.env file (delimiter `';'`):

```no-highlight
# Groups to search for user objects. "(sAMAccountName=%(user)s),..."
NAUTOBOT_AUTH_LDAP_USER_SEARCH_DN=OU=IT-Admins,OU=special-users,OU=Acme-User,DC=Acme,DC=local;OU=Infrastruktur,OU=IT,OU=my-location,OU=User,OU=Acme-User,DC=Acme,DC=local
```

Import LDAPSearchUnion in `nautobot_config.py`, and replace the AUTH_LDAP_USER_SEARCH command from above:

```python
from django_auth_ldap.config import ..., LDAPSearchUnion

# ...

user_search_dn_list = str(AUTH_LDAP_USER_SEARCH_DN).split(";")
ldapsearch_objects = []
for sdn in user_search_dn_list:
    ldapsearch_objects.append(LDAPSearch(sdn.strip(), ldap.SCOPE_SUBTREE, "(sAMAccountName=%(user)s)"))
AUTH_LDAP_USER_SEARCH = LDAPSearchUnion(*ldapsearch_objects)
```
````

**Review by glennmatthews — COMMENTED** [review]
Thanks for the PR!

**Comment by jifox** [comment]
I've fixed the render problem and added  os.getenv(... to example code.

**Review by glennmatthews — APPROVED** [review]
Thank you!

**Review by jathanism — APPROVED** [review]

**Pull request merged**